developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,221 @@
+output binary file. This config includes many module attributes such as name, uuid
+or capabilities. The host driver will rely on the extended module config for IPC
+messages to the DSP
+
+Rimage builds extended module config based on module setting in config/xxx.toml.
+The following setting is for gain module.
+
+.. code-block:: bash
+
+        [[module.entry]]
+        name = "GAIN"
+        uuid = "61BCA9A8-18D0-4A18-8E7B-2639219804B7"
+        affinity_mask = "0x1"
+        instance_count = "40"
+        domain_types = "0"
+        load_type = "0"
+        module_type = "5"
+        auto_start = "0"
+        init_config = "0"
+
+        # pin = [dir, type, sample rate, size, container, channel-cfg]
+        pin = [0, 0, 0xfeef, 0xf, 0xf, 0x45ff,
+                        1, 0, 0xfeef, 0xf, 0xf, 0x1ff]
+
+        # mod_cfg [PAR_0 PAR_1 PAR_2 PAR_3 IS_BYTES CPS IBS OBS MOD_FLAGS CPC OBLS]
+        mod_cfg = [0, 0, 0, 0, 416, 914000, 48, 64, 0, 0, 0,
+                   1, 0, 0, 0, 416, 1321600, 192, 256, 0, 0, 0,
+                   2, 0, 0, 0, 416, 1786000, 192, 256, 0, 0, 0,
+                   3, 0, 0, 0, 416, 2333000, 48, 64, 0, 0, 0,
+                   4, 0, 0, 0, 416, 2910000, 192, 256, 0, 0, 0,
+                   5, 0, 0, 0, 416, 3441000, 192, 256, 0, 0, 0,
+                   6, 0, 0, 0, 416, 4265000, 192, 256, 0, 0, 0]
+
+Module entry definition
+***********************
+| **name**: name of the module
+| **uuid**: uuid of the module
+| **affinity_mask**: restriction on which core a module can run, with the actual determination done on a per-instance basis.
+|                    bit N is for core N, 0x3 means module instance can run on core 0 and 1. Use 0xffff to ignore this restriction.
+| **instance_count**: max module instance count can be active in fw
+| **domain_types**: two schedule domains: domain ll - low latency (0), and dp - data processing (1)
+| **load_type**: build-in module (BIT(0)), loadable (BIT(1)) or both BIT(0) | BIT(1)
+| **module_type**: defined in the following enum module_type
+
+.. code-block:: bash
+
+  enum module_type {
+        basefw          = 0,
+        mixin           = 1,
+        mixout          = 2,
+        copier          = 3,
+        peakvol         = 4,
+        updwmix         = 5,
+        mux             = 6,
+        src             = 7,
+        wov             = 8,
+        fx              = 9,
+        aec             = 10,
+        kpb             = 11,
+        micselect       = 12,
+        fxf             = 13,
+        audclass        = 14,
+        fakecopier      = 15,
+        iodriver        = 16,
+        whm             = 17,
+        gdbstub         = 18,
+        sensing         = 19,
+        max             = 20,
+        invalid         = emax
+   };
+
+| **auto_start**: Indicates whether a instance of the module should be created at the base fw startup
+| **init_config**: 0 (only basic config in payload) or 1 (basic config with extension that contains pin format).
```

Comment:
pin format means nothing to non-sof developers, let alone most SOF developers... please describe or link to something more descriptive

Reply:
@mwasko @mmaka1 this PR may be better taken over by you guys as the manifest experts ? Need to make sure we provide all the relevant usage info.